From b6b91e908c7f508e82392ecfbffe520c8f0751b2 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 21 Nov 2019 11:11:44 +0100
Subject: [PATCH 1000/1239] arm64: mvebu: common: fix ramdisk address

The ramdisk address was overlapping the space for the kernel Image,
so shift it to 0x9000000. Also improve the get_ramfs variable,
so that it does not use the hardcoded address.

Change-Id: Ia673d4f6eeb23bd79649579ee8f5f9796e1165e0
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/19052
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Tested-by: Kostya Porotchkin <kostap@marvell.com>
---
 include/configs/mvebu_armada-common.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/mvebu_armada-common.h b/include/configs/mvebu_armada-common.h
index 4796fdd1c2..3d3174cb98 100644
--- a/include/configs/mvebu_armada-common.h
+++ b/include/configs/mvebu_armada-common.h
@@ -40,7 +40,7 @@
 					"loadaddr=0x6000000\0"		\
 					"fdt_high=0xffffffffffffffff\0"	\
 					"hostname=marvell\0"		\
-					"ramdisk_addr_r=0x8000000\0"	\
+					"ramdisk_addr_r=0x9000000\0"	\
 					"ramfs_name=-\0"		\
 					"cpuidle=cpuidle.off=1\0"	\
 					"fdt_name=fdt.dtb\0"		\
@@ -53,7 +53,7 @@
 					"image_name=Image\0"		\
 					"get_ramfs=if test \"${ramfs_name}\"" \
 						" != \"-\"; then setenv " \
-						"ramdisk_addr_r 0x8000000; " \
+						"ramdisk_addr_r $ramdisk_addr_r; " \
 						"tftpboot $ramdisk_addr_r " \
 						"$ramfs_name; else setenv " \
 						"ramdisk_addr_r -;fi\0"	\
-- 
2.29.0

